store/sqlstore/system_store.go

```diff
@@ -86,3 +88,37 @@ func (s SqlSystemStore) PermanentDeleteByName(name string) (*model.System, *mode
 
 	return &system, nil
 }
+
+// InsertIfExists inserts a given system value if it does not already exist. If a value
+// already exists, it returns the old one, else returns the new one.
+func (s SqlSystemStore) InsertIfExists(system *model.System) (*model.System, *model.AppError) {
+	tx, err := s.GetMaster().BeginTx(context.Background(), &sql.TxOptions{
+		Isolation: sql.LevelSerializable,
+	})
+	if err != nil {
+		return nil, model.NewAppError("SqlSystemStore.InsertIfExists", "store.sql_system.save.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	defer finalizeTransaction(tx)
+
+	var origSystem model.System
+	if err := tx.SelectOne(&origSystem, `SELECT * FROM Systems
+		WHERE Name = :Name`,
+		map[string]interface{}{"Name": system.Name}); err != nil && err != sql.ErrNoRows {
+		return nil, model.NewAppError("SqlSystemStore.InsertIfExists", "store.sql_system.get_by_name.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	if origSystem.Value != "" {
+		// Already a value exists, return that.
+		return &origSystem, nil
+	}
+
+	// Key does not exist, need to insert.
+	if err := tx.Insert(system); err != nil {
+		return nil, model.NewAppError("SqlSystemStore.InsertIfExists", "store.sql_system.save.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	if err := tx.Commit(); err != nil {
```

Comment:
Do we have/need a strategy in case the transaction were to fail? 
If this is something needed to have the server operating we might require some kind of re-trying mechanism.

Reply:
Discussed offline. Since this is a relatively simple get-update query, there's very low chances of the transaction failing. And in general, retrying failed transactions are something not done in our codebase as of now. Skipping this.